### ISSUE TITLE
`winget upgrade` lists an available update for WinSCP, but `winget upgrade WinSCP.WinSCP` says there's none

### ISSUE THREAD

**Issue by sba923** [microsoft/winget-cli#3054]
### Brief description of your issue

When enumerating available updates, `winget upgrade` says WinSCP 5.21.6 can be updated to 5.21.7.

But ` winget upgrade WinSCP.WinSCP` says there's no applicable update.

![image](https://user-images.githubusercontent.com/12860484/223431820-37cb42f1-05c8-4e40-8990-693ab031906b.png)

### Steps to reproduce

Run ` winget upgrade WinSCP.WinSCP` on a machine where WinSCP 5.21.6 is installed.

### Expected behavior

`winget` should upgrade WinSCP to 5.21.7

### Actual behavior

` winget upgrade WinSCP.WinSCP` reports there's no applicable update.

### Environment

```shell
Windows Package Manager v1.4.10173
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19045.2604
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.19.10173.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

User Settings: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by sba923** [comment]
Here's what I get from `winget list`:

```powershell
PS> winget list | sls -raw 'Name|winscp'
Name                                          Id                                             Version            Source
WinSCP 5.21.6                                 WinSCP.WinSCP                                  5.21.6             winget
```

Uninstalling will probably get rid of the problem, but isn't there something you'd like to look at / data you'd like to collect first?

**Comment by denelon** [comment]
Yes, since you only have one entry, please run `winget upgrade WinSCP.WinSCP --verbose-logs` and share the latest one in the issue (if you don't mind sharing). I'd like to see why WinGet doesn't think that an upgrade applies.

You can find the log location via `winget --info`.

**Comment by sba923** [comment]
This _should_ not contain anything too sensitive 😜

[WinGet-2023-03-07-19-21-45.413 - Copy.log](https://github.com/microsoft/winget-cli/files/10912798/WinGet-2023-03-07-19-21-45.413.-.Copy.log)

**Comment by denelon** [comment]
It's a different scope.

Let me know if:

`winget upgrade WinSCP.WinSCOP --scope user` works.

If it does, I need to find if we have that as a bug and if not, I'll create one.

```text
2023-03-07 19:21:50.909 [CLI ] Installer [X86,inno,Machine,en-US] not applicable: Installer scope does not match currently installed scope: Machine != User
```

**Comment by sba923** [comment]
```powershell
PS> winget upgrade WinSCP.WinSCP --scope user
No applicable upgrade found.
```

**Comment by denelon** [comment]
@Trenly

Have we seen this before? The manifest appears to have both scopes available.

https://github.com/microsoft/winget-pkgs/blob/master/manifests/w/WinSCP/WinSCP/5.21/WinSCP.WinSCP.installer.yaml

@sba923 are you wanting to stay with one scope vs. the other?

You could try using `winget install WinSCP.WinSCP --scope user --force`

Clearly, we have something not quite right here.

**Comment by denelon** [comment]
Thanks @ItzLevvie !!!

It looks like we need to add the missing scopes to the earlier manifests.

**Comment by sba923** [comment]
How do I determine the scope of the current install?

**Comment by denelon** [comment]
WinGet attempts to verify by looking to see if the registry keys are under HKCU or HKLM if I'm not mistaken.
